src/com/carolinarollergirls/scoreboard/defaults/DefaultScoreBoardModel.java

```diff
@@ -319,74 +315,58 @@ public void timeout(TeamModel team, boolean review) {
 		}
 	}
 
-	public void unStartJam() {
-		synchronized (runLock) {
-			if (!getClock(Clock.ID_JAM).isRunning())
-				return;
-
-			requestBatchStart();
-			if (lineupClockWasRunning)
-				getClockModel(Clock.ID_LINEUP).unstop();
-			if (timeoutClockWasRunning) {
-				setTimeoutOwner(lastTimeoutOwner);
-				setOfficialReview(wasOfficialReview);
-				getClockModel(Clock.ID_TIMEOUT).unstop();
+	protected void createSnapshot(String type) {
+		snapshot = new ScoreBoardSnapshot(this, type);
+	}
+	protected void restoreSnapshot() {
+		relapseTime = DefaultClockModel.updateClockTimerTask.getCurrentTime() - snapshot.getSnapshotTime();
+		for (ClockModel clock : getClockModels()) {
+			clock.restoreSnapshot(snapshot.getClockSnapshot(clock.getId()));
+		}
+		for (TeamModel team : getTeamModels()) {
+			team.restoreSnapshot(snapshot.getTeamSnapshot(team.getId()));
+		}
+		setTimeoutOwner(snapshot.getTimeoutOwner());
+		setOfficialReview(snapshot.isOfficialReview());
+		setInOvertime(snapshot.inOvertime());
+		setInPeriod(snapshot.inPeriod());
+		snapshot = null;
+	}
+	protected void relapseTime() {
+		for (ClockModel clock : getClockModels()) {
+			if (clock.isRunning()) {
+				clock.elapseTime(relapseTime);
```

Comment:
Passing replaseTime between these functions via a field of the object is a bit weird, it's creating an implicit dependency between restoreSnapshot and relapseTime. I think it'd be better to either have one call the other, or to merge the functions.

Reply:
replaseTime being a global-ish variable that's passing state behind the scenes is still weird, the dependency should be explicit.